### PR TITLE
Fix RCS1221: Use pattern matching instead of combination of 'as' operator and null check

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Autofix RCS1221

## PR Context

* #13287
* #13323

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [NA] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [NA] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [NA] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [NA] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [NA] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/powershell/13333)

### CODE REVIEW

**Comment by xtqqczze** [comment]
Reverted IDE0019 commits due to high number of issues.

**Comment by xtqqczze** [comment]
PR is ready for review

**Comment by xtqqczze** [comment]
@iSazonov Can you review?

File: src/System.Management.Automation/help/ProviderContext.cs [hunk 1]
```diff
@@ -108,10 +108,7 @@ internal MamlCommandHelpInfo GetProviderSpecificHelpInfo(string helpItemName)
 
             // Does the provider know how to generate MAML.
             CmdletProvider cmdletProvider = providerInfo.CreateInstance();
-            ICmdletProviderSupportsHelp provider = cmdletProvider as ICmdletProviderSupportsHelp;
-
-            // Under JEA sessions the resolvedProviderPath will be null, we should allow get-help to continue.
-            if (provider == null)
```
Comment (iSazonov) [inline src/System.Management.Automation/help/ProviderContext.cs:114]: We lost the comment. Please restore.

**Review by iSazonov — APPROVED** [review]
LGTM with one comment.

**Comment by daxian-dbw** [comment]
@PowerShell/powershell-maintainers reviewed this PR and agreed to accept it for 7.2-preivew. So we should delay merging it until after 7.1-RC1 release branch gets created.
> The changes make sense, but it requires line-by-line review to make sure there is no potential incorrect change, so it's better to delay merging it till 7.2 preview to avoid possible regression.

**Comment by iSazonov** [comment]
Set Draft till 7.2. Preview1.

**Comment by xtqqczze** [comment]
@rjmholt Could you please restart tests.

**Comment by iSazonov** [comment]
I restarted CI Windows.

**Comment by xtqqczze** [comment]
@iSazonov Please restart `PowerShell-CI-windows` another time.

**Comment by iSazonov** [comment]
Restarted CI Windows.

**Comment by xtqqczze** [comment]
@iSazonov It seems Github is still reporting the old `PowerShell-CI-windows` result. I shall rerun tests by rebasing instead.

**Comment by xtqqczze** [comment]
@iSazonov Should I fix pre-existing Codacy issues : `Add curly braces around the nested statement(s) in this 'if' block.`?

**Comment by iSazonov** [comment]
> @iSazonov Should I fix pre-existing Codacy issues : `Add curly braces around the nested statement(s) in this 'if' block.`?

In follow PR only.

**Comment by xtqqczze** [comment]
This PR is ready for maintainer review.

**Comment by xtqqczze** [comment]
@iSazonov I am considering closing in favour of #13895 because codefix for IDE0019:
* automatically uses `is not` syntax (so no manual changes required)
* does not have any problems with stripped comments

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
> * Use `is not` syntax (like #13323, every change has been manually reviewed)

I don't see the change in the PR but we can do this later and I am ready to merge.

**Comment by xtqqczze** [comment]
> > * Use `is not` syntax (like #13323, every change has been manually reviewed)
> 
> I don't see the change in the PR but we can do this later and I am ready to merge.

I think this change was removed in a rebase to refocus the PR. Further `is not` changes can be made in follow-up PR. So please merge this PR.

**Comment by iSazonov** [comment]
@xtqqczze Do you think about forcing Roslynator rules? We could add new GitHub Check.

**Comment by xtqqczze** [comment]
@iSazonov I used Roslynator to make these changes as it was the easiest tool for me at the time of the PR. Now codestyle analyzers are included in .NET 5.0 we can use [IDE0019](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0019) instead.

**Comment by iSazonov** [comment]
> Now codestyle analyzers are included in .NET 5.0

All?

**Comment by xtqqczze** [comment]
> > Now codestyle analyzers are included in .NET 5.0
> 
> All?

Many RCSXXXX rules have crossover with IDEXXXX and CAXXXX. If we also add StyleCop analyzers package like in #13963 we can likely cover most scenarios.

**Pull request merged**